h/templates/embed.js

```diff
@@ -46,6 +46,17 @@
       resources = resources.concat(['{{ layout.xpath_polyfill_urls | map("string") | join("', '") | safe }}']);
     }
 
+    // https://github.com/Modernizr/Modernizr/blob/master/feature-detects/url/parser.js
+    var url;
+    try {
+      // have to actually try use it, because Safari defines a dud constructor
+      url = new URL('http://modernizr.com/');
+      var works = url.href === 'http://modernizr.com/';
```

Comment:
In Modernizr this is used as a test and returns a boolean. Notice how the exception case returns `false` but otherwise the comparison result is returned.

Unless you wrap this up as a test function and use that in a conditional, we're not actually catching the case where the constructor works because `works` is never checked. If `works` is `false` we want the polyfill.